tests/imapp_benchmark/imapp_bench.go

```diff
@@ -0,0 +1,249 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"math"
+	"math/big"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	_ "unsafe"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/rawdb"
+	"github.com/ethereum/go-ethereum/core/state"
+	"github.com/ethereum/go-ethereum/core/vm/runtime"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/params"
+)
+
+var result []byte
+var bytecode []byte
+var bytecodeStore string
+
+func BenchmarkBytecodeExecutionSerial(b *testing.B) {
+	var calldata []byte
+
+	cfg := new(runtime.Config)
+	setDefaults(cfg)
+	// from `github.com/ethereum/go-ethereum/core/vm/runtime/runtime.go:109`
+	cfg.State, _ = state.New(common.Hash{}, state.NewDatabase(rawdb.NewMemoryDatabase()), nil)
+
+	// Initialize some constant calldata of 128KB, 2^17 bytes.
+	// This means, if we offset between 0th and 2^16th byte, we can fetch between 0 and 2^16 bytes (64KB)
+	// In consequence, we need args to memory-copying OPCODEs to be between 0 and 2^16, 2^16 fits in a PUSH3,
+	// which we'll be using to generate arguments for those OPCODEs.
+	calldata = []byte(strings.Repeat("{", 1<<17))
+
+	// Warm-up. **NOTE** we're keeping tracing on during warm-up, otherwise measurements are off
+	cfg.EVMConfig.Debug = false
+	_, _, errWarmUp := runtime.Execute(bytecode, calldata, cfg)
```

Comment:
same comment here, I thought we'd be removing this home-grown warm-up?

Reply:
Done